pkg/controller/node/nodecontroller.go

```diff
@@ -375,7 +375,7 @@ func (nc *NodeController) allocateOrOccupyCIDR(obj interface{}) {
 	glog.V(4).Infof("Assigning node %s CIDR %s", node.Name, podCIDR)
 	for rep := 0; rep < podCIDRUpdateRetry; rep++ {
 		node.Spec.PodCIDR = podCIDR.String()
-		if _, err := nc.kubeClient.Core().Nodes().Update(node); err == nil {
+		if _, err := nc.kubeClient.Core().Nodes().Update(node); err != nil {
 			glog.Errorf("Failed while updating Node.Spec.PodCIDR : %v", err)
```

Comment:
hmm, if i run into error on update (like resource version conflict) I'd want to re-get and retry the update till podCIDRUpdateRetry. This will break on first error?